internal/app/machined/pkg/runtime/v1alpha1/bootloader/grub/boot_label.go

```diff
@@ -0,0 +1,38 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package grub
+
+import (
+	"fmt"
+	"strings"
+)
+
+// BootLabel represents a boot label, e.g. A or B.
+type BootLabel string
+
+// FlipBootLabel flips the boot entry, e.g. A -> B, B -> A.
+func FlipBootLabel(e BootLabel) (BootLabel, error) {
+	switch e {
+	case BootA:
+		return BootB, nil
+	case BootB:
+		return BootA, nil
+	default:
+		return "", fmt.Errorf("invalid entry: %s", e)
+	}
+}
+
+// ParseBootLabel parses the given human-readable boot label to a grub.BootLabel.
+func ParseBootLabel(name string) (BootLabel, error) {
+	if strings.HasPrefix(name, string(BootA)) {
```

Comment:
These prefixes are pretty broad.  I know we should own the bootloader config, but this potentially matches a lot of stuff.  For instance, it would match `Arch Linux` or `Boot into Memtest86+`.

Could we include the punctuation here, as well:  `A - `

Reply:
we can't include punctuation, as currently descriptions (labels) are `A` and `B` (this will break upgrade from current Talos)

we could make it _more_ smart, but I guess it doesn't give us much, as Talos never preserved old grub config really, it always overwrites it.

Reply:
There seems to already bepunctuation:  It's "A - blah blah" and "B - blah blah".  Am I missing something?

Reply:
I mean Talos before this PR does simply `A`. So if one upgrades to the next version, it should still treat `A` as `BootLabelA`, and it will add `B - v1.2.0`, but `A` should be preserved as it was (as previous version before the upgrade)

Reply:
Ah, yes.  Good point. :+1: